lib/wallet/mod.rs

```diff
@@ -1931,4 +1931,280 @@ impl Wallet {
     ) -> Result<(), miette::Report> {
         self.inner.create_new_wallet(mnemonic, password).await
     }
+
+    /// Create a multisig address requiring multiple signatures to spend from.
+    ///
+    /// # Arguments
+    /// * `required_signatures` - Number of signatures required (m in m-of-n)
+    /// * `public_keys` - List of public keys in hex format
+    /// * `address_type` - Type of address to create: "p2sh", "p2wsh", or "p2sh-p2wsh"
+    ///
+    /// # Returns
+    /// Returns a tuple containing:
+    /// * The multisig address
+    /// * The redeem script in hex
+    /// * The output descriptor
+    pub fn create_multisig_address(
+        &self,
+        required_signatures: u32,
+        public_keys: &[String],
+        address_type: Option<&str>,
+    ) -> Result<(bitcoin::Address, String, String)> {
```

Comment:
I'm not sure the correct place for this is in the enforcer /wallet/. It doesn't actually use any parts of the wallet, from what I can tell. Would the validator be a better place?

Reply:
We should also not be returning strings. For the descriptor, we should be returning a properly typed descriptor from the `miniscript` crate: https://docs.rs/miniscript/12.3.0/miniscript/index.html#output-descriptors. It is re-exported by `bdk_wallet`. For the redeem script, what about a `ScriptBuf`?